AMM/SPECIFICATION.md

```diff
@@ -0,0 +1,180 @@
+    1. If the asset pair of the pool is set 
+    2. Requires any amount of either asset in the pair
+
+#### `add_liquidity()`
+
+1. Allows minting liquidity pool assets by using up the deposits when possible 
+    1. If
+        1. The asset pair of the pool is set
+        2. The deadline has not passed
+        3. Deposit amounts of both assets are sufficient
+        4. The resulting liquidity amount is more than the minimum liquidity for a pool to exist. 
+            > **NOTE** This lower limit for liquidity can be modified as desired per asset pair and exists to protect against excessive slippage.
+    2. Requires 
+        1. The desired liquidity amount
+        2. A deadline, i.e., block height limit
+
+#### `remove_liquidity()`
+
+1. Allows burning liquidity pool assets Requires 
+    1. If
+        1. The asset pair of the pool is set
+        2. The deadline has not passed
+        3. The amount of liquidity pool assets are sufficient
+    2. Requires
+        1. The minimum amounts of both assets to receive after burning
+        2. A deadline, i.e., block height limit
+
+#### `withdraw()`
+
+1. Allows withdrawing previously deposited assets without using them to add liquidity 
+    1. If
+        1. The asset pair of the pool is set
+        2. The deposited amount of the asset is sufficient
+    2. Requires an amount of either asset to withdraw
+
+#### `swap_with_exact_input()`
+
+1. Allows selling an exact amount of an asset for the other asset 
+    1. If 
+        1. The asset pair of the pool is set
+        2. The deadline has not passed
+        3. The asset reserve of asset to buy in the contract is sufficient
+        4. The bought amount is not less than the optional minimum desired output amount
+    2. Requires
+        1. An exact amount of either asset to sell
+        2. An optional minimum desired output amount
+            > **NOTE** This is a safety mechanism against excessive slippage. The [`preview_swap_with_exact_input()`](#preview_swap_with_exact_input) function can be used to calculate a reasonable minimum output amount.
+        3. A deadline, i.e., block height limit
+
+
+#### `swap_with_exact_output()`
+1. Allows selling an asset for an exact amount of the other asset 
+    1. If
+        1. The asset pair of the pool is set
+        2. The deadline has not passed
+        3. The input amount is sufficient
+        4. The asset reserve of asset to buy in the contract is sufficient
+        5. The bought amount is not less than the exact output amount
+    2. Requires
+        1. An exact amount of either asset to buy
+        2. A maximum input amount 
+            > **NOTE** This is a safety mechanism against excessive slippage. The [`preview_swap_with_exact_output()`](#preview_swap_with_exact_output) function can be used to calculate a reasonable maximum input amount.    
+        3. A deadline, i.e., block height limit
+
+### Exchange Previews
```

Comment:
Similar comments for specificity

Reply:
Do you mean "other asset" specificity or "minimum / maximum / input / output / optional" specificity here?

Reply:
Generally anything that can be specific to remove any inferences the better